crates/jit/src/compiler.rs

```diff
@@ -166,9 +166,16 @@ impl Compiler {
             let body_len = compilation.get(i).body.len();
             self.jit_function_ranges
                 .push((ptr as usize, ptr as usize + body_len));
+            let func_index = module.func_index(i);
+            let func_name = module
+                .func_names
+                .get(func_index)
+                .cloned()
+                .unwrap_or_else(Default::default);
```

Comment:
I think this may be better as threading through the `Option` to explicitly denote functions in the wasm module that didn't have a name listed in the name section.

Reply:
[Rust's doc](https://doc.rust-lang.org/std/string/struct.String.html#method.new) says that String do not allocate buffer for empty string -- thought it will be preferable for this purpose.

Reply:
It's true that it doesn't allocate, but I think there's an important semantic difference between "no function name listed" vs "function name listed but it was an empty string"

Reply:
> "no function name listed" vs "function name listed but it was an empty string"

`SecondaryMap` is used, which defaults everything to empty string.

Be aware that name section has reputation of being unreliable, which can be absent, invalid or incomplete, so difference between empty/non-defined a little bit blurred here.

Reply:
Hm I'm not sure about the limitations of `SecondaryMap`, but at least in the API of the `wasmtime` crate I think we'll want to return an optional string and we can figure out the implementation details later if necessary.